dallinger/db.py

```diff
@@ -154,6 +155,91 @@ def init_db(drop_all=False, bind=engine):
     return session
 
 
+def get_all_mapped_classes():
+    """
+    Lists the different classes that are mapped with SQLAlchemy.
+    Classes are only included if they have at least one row in the database.
+    Returns a dictionary, keyed by class names,
+    where each value is itself a dictionary with three values:
+    ``cls``, the class itself;
+    ``table``, the database table within which the class can be found,
+    and ``polymorphic_identity``, the string label with which the class is
+    identified in the table's ``type`` column. The ``polymorphic_identity`` field
+    takes a value of ``None`` if the table does not use polymorphic inheritance.
+    """
+    classes = {}
+    for table in Base.metadata.tables.values():
+        if "type" in table.columns:
```

Comment:
Can you elaborate on this if else statement? When does it happen that type is not a column in the table?

Reply:
I have now added some comments to the code explaining this a bit more.

```
            # Most Dallinger tables (e.g. Node, Network) have a type column that specifies which class
            # is associated with that database row.
            # Some tables (e.g. Notification) don't have any such column, so we can assume
            # that they have exactly one mapped class.
```

Reply:
That makes sense! Thanks for elaborating!